CONTRIBUTING.md

```diff
@@ -1,4 +1,134 @@
 # Contributing to Theatre.js
 
->`@todo` Write the contribution guide.
+## Getting Started
 
+### Prerequisites
+
+Make sure you have [Node](https://nodejs.org/) installed at v14.0.0+
+
+```sh
+node -v
+
+v14.0.0
+
+```
+
+and [Yarn](https://classic.yarnpkg.com/en/) at v2.0.0+.
+
+```sh
+yarn -v
+
+2.0.0
+```
+
+### Fork, Clone & Install
+
+Start by forking Theatre.js to your GitHub account. Then clone your fork and
+install dependencies:
+
+```sh
+git clone git@github.com:<your-user>/theatre.git
+cd theatre
+yarn
+```
+
+> ⚠ theatre relies on
+> [yarn workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces/) so
+> `npm install` will not work in this repository.
+
+Add our repo as a git remote so you can pull/rebase your fork with our latest
+updates:
+
+```
+git remote add upstream git@github.com:AriaMinaei/theatre.git
+```
+
+## Project structure
+
+> This section is lacking in instructions (PRs welcome!).
+
+### `@theatre/core`
+
+The core animation library. This project is located under
+[`./theatre/core`](`/theatre/core).
+
+### `@theatre/studio`
+
+The visual editor. This project is located under
+[`./theatre/studio`](`/theatre/studio).
+
+### Commands
+
+> This list is not updated, you should run `yarn run` to see all scripts.
+
+#### Root commands
+
+```sh
+yarn playground            // run the playground
+
+yarn test                  // run all tests
+yarn typecheck             // TS typechecking
+```
+
+#### Theatre workspace commands
+
+Commands available for the `@theatre` workspace:
+
+```sh
+yarn build:js
+yarn build:js:watch
+```
+
+## Devflow
+
+> This section is lacking in instructions (PRs welcome!).
+
+When working on changes for the `@theatre` workspace, you want to have a couple
+of things running concurrently:
+
+1. Run the playground to verify the changes
+   ```
+   yarn playground
+   ```
+2. Watch and rebuild theatre packages on changes:
+   ```sh
+   cd theatre
+   yarn build:js
+   ```
```

Comment:
Not entirely sure about this? It seems to be working for me. If I add a console log inside the `@theatre/studio`, it does appear after i refresh the playground 👍 

However the `playground` is just a blank screen with the studio and a couple of console errors.

Reply:
![image](https://user-images.githubusercontent.com/837651/135693025-3f4a43cb-c5e8-48e3-8327-8e7087835d3a.png)

Reply:
Ah so playground is made so you don't have to build any of the packages. It has an esbuild config that automatically includes all the packages. So, if you want to get started hacking, you can just `yarn playground` and that should be enough.

Reply:
However, I think the stuff in `packages/playground/src` is a bit confusing. Like, what's supposed to happen with any of those playgrounds directories?

I think the goal of playground should be like this:
1. It's a quick cycle way to hack on theatre. No long builds, no wait time, just a single esbuild setup for all the packages.
2. `packages/playground/src/index.ts` just imports one of the playground directories, like so: `import './shared/audio/index.ts` or something like that. This `playground/src/index.ts` file is actually not committed to the repo. It will be created first time you run `yarn playground` and is `.gitignore`d. This allows you to change its `import` and for example, say `import './local/walk-cycle/index.ts`, without having to commit `src/index.ts` or `src/local` to the repo.
3. We'll have some ready-made playgrounds to in `packages/playground/src/shared/`, which _will_ be comitted to the repo.

Reply:
I'll do that playground thing now, so we can get the PR merged soon.

Reply:
Ok, playground is done.